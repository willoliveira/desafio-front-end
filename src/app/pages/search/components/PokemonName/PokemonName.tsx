import React, { FC } from 'react';
import { View, StyleSheet, Text } from 'react-native';

export interface IPokemonNameProps {
	pokemonNumber: number;
	name: string;
}

export const PokemonName: FC<IPokemonNameProps> = ({ pokemonNumber, name }: IPokemonNameProps) => (
	<View style={styles.pokemonNameContainer}>
		<View style={styles.nameContainer}>
			<Text style={styles.textName}>{getName(name)}</Text>
			<Text style={styles.textNumber}>{getNumber(pokemonNumber)}</Text>
		</View>
	</View>
);

export const getName = (name: string): string => `${name[0].toUpperCase()}${name.substring(1)}`;

export const getNumber = (pokemonNumber: number): string => {
	let number = `${pokemonNumber}`
	if (pokemonNumber.toString().length === 1) {
		number = `00${number}`;
	} else if (pokemonNumber.toString().length === 2) {
		number = `0${number}`;
	}

	return `NÂº${number}`;
}

const styles = StyleSheet.create({
	pokemonNameContainer: {
		width: '100%',
		marginBottom: 16,
		marginTop: 16
	},
	nameContainer: {
		flex: 1,
		flexDirection: 'row',
		alignContent: 'center',
	},
	textName: {
		fontSize: 20,
	},
	textNumber: {
		fontSize: 20,
		marginLeft: 4,
		color: '#616161'
	}
})
