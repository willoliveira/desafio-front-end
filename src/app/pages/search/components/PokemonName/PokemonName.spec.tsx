/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { PokemonName, getName, getNumber } from './PokemonName';

describe('<PokemonName />', () => {

	describe('Rendering', () => {

		it('should match to snapshot - Bulbassaur Nº001', () => {
			const component = shallow(<PokemonName name='Bulbassaur' pokemonNumber={1} />);
			expect(component).toMatchSnapshot("PokemonName snapshot");
		});

		it('should match to snapshot - Pikachu Nº025', () => {
			const component = shallow(<PokemonName name='Bulbassaur' pokemonNumber={1} />);
			expect(component).toMatchSnapshot("PokemonName snapshot");
		});

		it('should match to snapshot - Mewtwo Nº150', () => {
			const component = shallow(<PokemonName name='Bulbassaur' pokemonNumber={1} />);
			expect(component).toMatchSnapshot("PokemonName snapshot");
		});
	});

	describe('getName()', () => {

		it('CamelCase correct', () => {
			expect(getName('bulbassaur')).toEqual('Bulbassaur');
		});
	});

	describe('getNumber()', () => {

		it('Format number one digits - 1', () => {
			expect(getNumber(1)).toEqual('Nº001');
		});

		it('Format number two digits - 25', () => {
			expect(getNumber(25)).toEqual('Nº025');
		});

		it('Format number three digits - 125', () => {
			expect(getNumber(150)).toEqual('Nº150');
		});
	});
});
