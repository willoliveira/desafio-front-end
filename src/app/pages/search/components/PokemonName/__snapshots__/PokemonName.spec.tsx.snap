// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonName /> Rendering should match to snapshot - Bulbassaur Nº001: PokemonName snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
      "marginTop": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Bulbassaur
    </Text>
    <Text
      style={
        Object {
          "color": "#616161",
          "fontSize": 20,
          "marginLeft": 4,
        }
      }
    >
      Nº001
    </Text>
  </View>
</View>
`;

exports[`<PokemonName /> Rendering should match to snapshot - Mewtwo Nº150: PokemonName snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
      "marginTop": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Bulbassaur
    </Text>
    <Text
      style={
        Object {
          "color": "#616161",
          "fontSize": 20,
          "marginLeft": 4,
        }
      }
    >
      Nº001
    </Text>
  </View>
</View>
`;

exports[`<PokemonName /> Rendering should match to snapshot - Pikachu Nº025: PokemonName snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
      "marginTop": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Bulbassaur
    </Text>
    <Text
      style={
        Object {
          "color": "#616161",
          "fontSize": 20,
          "marginLeft": 4,
        }
      }
    >
      Nº001
    </Text>
  </View>
</View>
`;
